src/Domain/Encryption/Crypter.ts

```diff
@@ -0,0 +1,46 @@
+import { injectable } from 'inversify'
+import { randomBytes, createCipheriv, createDecipheriv } from 'crypto'
+
+import { CrypterInterface } from './CrypterInterface'
+
+@injectable()
+export class Crypter implements CrypterInterface {
+  static readonly ENCRYPTION_VERSION_1 = 1
+
+  async encrypt(encryptionVersion: number, data: string, masterKey: string): Promise<string> {
+    if (encryptionVersion !== Crypter.ENCRYPTION_VERSION_1) {
+      throw new Error(`Supported versions of encryption: ${Crypter.ENCRYPTION_VERSION_1}`)
+    }
+
+    const iv = randomBytes(16)
+
+    const salt = randomBytes(64)
+
+    const cipher = createCipheriv('aes-256-gcm', masterKey, iv)
+
+    const encrypted = Buffer.concat([cipher.update(data, 'utf8'), cipher.final()])
+
+    const tag = cipher.getAuthTag()
+
+    return Buffer.concat([salt, iv, tag, encrypted]).toString('base64')
+  }
+
+  async decrypt(encryptionVersion: number, data: string, masterKey: string): Promise<string> {
+    if (encryptionVersion !== Crypter.ENCRYPTION_VERSION_1) {
+      throw new Error(`Supported versions of encryption: ${Crypter.ENCRYPTION_VERSION_1}`)
+    }
+
+    const bData = Buffer.from(data, 'base64')
+
+    const iv = bData.slice(64, 80)
```

Comment:
What does bData[0, 64] represent then?

Reply:
the salt - the length of which should probably come from env vars to make it hard to guess for someone where the encrypted data starts

Reply:
> make it hard to guess for someone where the encrypted data starts
This isn't a real thing.

Reply:
What do you mean by it's not a real thing?

Example:

`U0Znam5UNUNsMGMzTlBlejdiUXJOTnNJa21zYlp5WWhBd3hMQ3lPN1RsT0QvYThTZWkzZUpaOG5LS1E5K3c3dFJMcytreWVmK3E3MmVrcjZ3TUdXQnJDCXbxJnpQXX+tUQDb/hN8ARUjZifwyrgfY0PGeuXV3slwI03O`

I can't see where the encrypted data starts

`U0Znam5UNUNsMGMzTlBlejdiUXJOT:nNJa21zYlp5WWhBd3hMQ3lPN1RsT0QvYThTZW:kzZUpaOG5LS1E5K3c3dFJMcytreWVmK3E3MmVrcjZ3TUdXQ:nJDCXbxJnpQXX+tUQDb/hN8ARUjZifwyrgfY0PGeuXV3slwI03O`

I can clearly see which block is the salt, the iv, the auth tag and the encrypted data

Reply:
The salt and IV are public information :) It's ok that you can see where they start/end.